backend/Room.js

```diff
@@ -1,138 +1,164 @@
 const utils = require('./utils')
 const _ = require('lodash');
 
-function Room(id){
-    this.name = "";
-    this.id = id;
-    this.users = [];
+function Room(id) {
+  this.name = "";
+  this.id = id;
+  this.players = [];
+  this.words = [];
+  this.wordsOfRound = [];
+  this.teams = [];
+  this.hasAGameMaster = false;
+  this.round = 0;
+  this.set = 0;
+  this.setFinished = false;
+  this.scoreFirstTeam = 0;
+  this.scoreSecondTeam = 0;
+  this.numberOfPlayer = 0;
+  this.players = [];
+  this.lastActivity = Date.now();
+}
+
+Room.prototype = {
+  getId: function() {
+    this.updateActivity();
+    return this.id;
+  },
+  getPlayers: function() {
+    this.updateActivity();
+    return this.players;
+  },
+  getPlayers: function() {
+    this.updateActivity();
+    return this.players;
+  },
+  getWordsOfRound: function() {
+    this.updateActivity();
+    return this.wordsOfRound;
+  },
+  getSet: function() {
+    this.updateActivity();
+    return this.set;
+  },
+  getTeams: function() {
+    this.updateActivity();
+    return this.teams;
+  },
+  isSetFinished: function() {
+    this.updateActivity();
+    return this.setFinished;
+  },
+  getScoreFirstTeam: function() {
+    this.updateActivity();
+    return this.scoreFirstTeam;
+  },
+  getScoreSecondTeam: function() {
+    this.updateActivity();
+    return this.scoreSecondTeam;
+  },
+  setName: function(name) {
+    this.updateActivity();
+    this.name = name;
+  },
+  setGameMaster: function() {
+    this.updateActivity();
+    this.hasAGameMaster = true;
+  },
+  addPlayer: function(player) {
+    this.updateActivity();
+    this.players = [
+      ...this.players,
+      player
+    ];
+    this.numberOfPlayer = this.players.length;
+  },
+  removePlayer: function(name) {
```

Comment:
This function misses to update the property `numberOfPlayer `.
However, maybe we could get rid of this property and use a method that returns `this.players.length` instead.